db/sql/0_msar.sql

```diff
@@ -130,16 +130,48 @@ $$ LANGUAGE plpgsql RETURNS NULL ON NULL INPUT;
 
 CREATE OR REPLACE FUNCTION
 msar.get_column_name(table_id oid, col_attnum integer) RETURNS text AS $$/*
-Return the name for a given table, qualified or quoted as appropriate.
+Return the name for a given column in a given table.
+
+Args:
+  table_id:  The OID of the table.
+  column_attnum:   The attnum of the column in the table.
+*/
+BEGIN
+  RETURN quote_ident(attname::text) FROM pg_attribute WHERE attrelid=table_id AND attnum=col_attnum;
+END;
+$$ LANGUAGE plpgsql RETURNS NULL ON NULL INPUT;
 
-In cases where the table is already included in the search path, the returned name will not be
-fully-qualified.
+
+CREATE OR REPLACE FUNCTION
+msar.is_pkey_col(table_id oid, col_attnum integer) RETURNS boolean AS $$/*
+Return whether the given column is in the primary key of the given table.
+*/
+BEGIN
+  RETURN ARRAY[col_attnum::smallint] <@ conkey FROM pg_constraint WHERE conrelid=table_id;
+END;
+$$ LANGUAGE plpgsql RETURNS NULL ON NULL INPUT;
+
+
+CREATE OR REPLACE FUNCTION
+msar.get_cast_function_name(target_type regtype) RETURNS text AS $$/*
+Return a string giving the appropriate name of the casting function for the target_type.
 
 Args:
-  table_id:   The OID of the table.
+  target_type: This should be a type that exists.
 */
+DECLARE target_type_prepped text;
 BEGIN
-  RETURN attname::text FROM pg_attribute WHERE attrelid=table_id AND attnum=col_attnum;
+  WITH unqualifier AS (
+    SELECT x[array_upper(x, 1)] unqualified_type
+    FROM regexp_split_to_array(target_type::text, '\.') x
+  ), unspacer AS(
+    SELECT replace(unqualified_type, ' ', '_') unspaced_type
+    FROM unqualifier
+  )
+  SELECT replace(unspaced_type, '"', '_double_quote_')
+  FROM unspacer
+  INTO target_type_prepped;
+  RETURN format('mathesar_types.cast_to_%s', target_type_prepped);
 END;
 $$ LANGUAGE plpgsql RETURNS NULL ON NULL INPUT;
```

Comment:
What do you think about including in the docstrings some commentary about the algorithm? Some of these functions are not easy to parse.

Reply:
We can do it where needed, but I prefer to minimize the number of places where we need to maintain the same algorithm. 

That said, I've expanded the docstring for this function, as it is a bit difficult.